### PR TITLE
#2/Create-a-Meal-Plan

### PR DESCRIPTION
**Pull request by omorgenroth**
Details in the [User Story](https://github.com/omorgenroth/capstone-project/issues/2)

### CODE REVIEW

File: frontend/src/App.js [hunk 1]
```diff
@@ -1,31 +1,61 @@
...
+          error={error}
+        />
+      </Route>
     </div>
   )
 
-  function addIsCheckedValue(data) {
-    return data.map((item) => ({ ...item, isChecked: true }))
+  function addIsSelectedValue(data) {
+    return data.map((item) => ({ ...item, isSelected: false }))
```
Comment (FabianAlleblas) [inline frontend/src/App.js:58]: What is the selected value? and What is data?
Would rename some stuff to be more specific about it...

File: frontend/src/Components/HeaderOverlay/HeaderOverlay.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import PropTypes from 'prop-types'
+import { Link } from 'react-router-dom'
+
+HeaderOverlay.propTypes = {
+  children: PropTypes.string,
+}
+
+export default function HeaderOverlay({ children }) {
+  return (
+    <HeaderStyled>
```
Comment (FabianAlleblas) [inline frontend/src/Components/HeaderOverlay/HeaderOverlay.js:11]: Would rename it to Header

File: frontend/src/Pages/DishOverviewPage.js [hunk 3]
```diff
@@ -1,46 +1,95 @@
...
+}
+
+export default function DishOverview({ dishes, onToggleItem, error }) {
+  let counter = dishes.filter((dish) => {
+    return dish.isSelected
+  }).length
 
-export default function DishOverview({ dishes }) {
   return (
     <Wrapper>
```
Comment (FabianAlleblas) [inline frontend/src/Pages/DishOverviewPage.js:19]: Wrapper with grid, grid with grid.... naming 😅

**Review by FabianAlleblas — COMMENTED** [review]
Nice one 🚀 , your CSS looks a bit overwhelming and your naming could need some work...

**Review by felix-horn — COMMENTED** [review]
This is only the first part of my review. I got distracted :) Tomorrow I'll do the rest. What I have seen so far, I liked!!

File: frontend/src/App.js [hunk 4]
```diff
@@ -1,31 +1,61 @@
...
     <div className="App">
-      <Switch>
-        <Route exact path="/">
-          <LandingPage />
-        </Route>
-        <Route path="/dishes">
-          <DishOverviewPage dishes={dishes} />
-        </Route>
-      </Switch>
+      <Route exact path="/">
```
Comment (ninapeters) [inline frontend/src/App.js:33]: Why did you remove the `Switch`?

File: frontend/src/Components/DishItem/DishItem.js [hunk 5]
```diff
@@ -3,42 +3,46 @@ import styled from 'styled-components/macro'
 import PropTypes from 'prop-types'
 
 DishItem.propTypes = {
-  name: PropTypes.string.isRequired,
-  isChecked: PropTypes.bool,
+  id: PropTypes.number,
+  title: PropTypes.string,
+  isSelected: PropTypes.bool,
+  onSelect: PropTypes.func,
```
Comment (ninapeters) [inline frontend/src/Components/DishItem/DishItem.js:9]: Is any of this required?

File: frontend/src/Components/DishItem/DishItem.js [hunk 6]
```diff
@@ -3,42 +3,46 @@ import styled from 'styled-components/macro'
...
 
-export default function DishItem({ name, isChecked }) {
+export default function DishItem({ id, title, isSelected, onClick }) {
   return (
-    <ItemWrapper>
+    <ItemWrapper onClick={() => onClick(id)} selected={isSelected}>
       <ContentWrapper>
-        <Title> {name}</Title>
-        <p> Ingredient1, Ingredient2, Ingredient3 </p>
+        <Title> {title}</Title>
```
Comment (ninapeters) [inline frontend/src/Components/DishItem/DishItem.js:16]: I don't think you need the space.

File: frontend/src/Components/DishItem/DishItem.js [hunk 7]
```diff
@@ -3,42 +3,46 @@ import styled from 'styled-components/macro'
...
     </ItemWrapper>
   )
 }
 
 const ItemWrapper = styled.div`
-  background-color: ${(props) => (props.checked ? 'blue' : 'var(--c-white)')};
-  box-shadow: 0, 3px, 6px, rgba(0, 0, 0, 0.5);
+  background-color: ${(props) =>
+    props.selected ? 'var(--c-green)' : 'var(--c-white)'};
+  box-shadow: 0px 0px 6px 3px rgba(0, 0, 0, 0.03);
```
Comment (ninapeters) [inline frontend/src/Components/DishItem/DishItem.js:32]: You should also use a color varibale.

File: frontend/src/Components/DishItem/DishItem.js [hunk 8]
```diff
@@ -3,42 +3,46 @@ import styled from 'styled-components/macro'
...
   display: grid;
   grid-template-columns: 3fr 1fr;
-  width: 85%;
+  width: 100%;
   height: 60px;
   position: relative;
 `
-const Title = styled.h2`
-  font-size: 1.5em;
+const Title = styled.p`
```
Comment (ninapeters) [inline frontend/src/Components/DishItem/DishItem.js:44]: Is that `p` semantically correct? Maybe a `h` is better like it was before.

File: frontend/src/Components/DishItem/DishItem.js [hunk 9]
```diff
@@ -51,14 +55,14 @@ const IconWrapper = styled.div`
 
 const CheckedIcon = styled(FaCheckCircle)`
   color: var(--c-gray);
-  transform: scale(2);
+  transform: scale(2.5);
   position: absolute;
```
Comment (ninapeters) [inline frontend/src/Components/DishItem/DishItem.js:59]: Do you really want to position the icon? It may be better to position it over a parent element. Have you checked it in your styleguide?

File: frontend/src/Components/DishItem/DishItem.js [hunk 10]
```diff
@@ -51,14 +55,14 @@ const IconWrapper = styled.div`
...
   top: 25px;
   right: 30px;
 `
 const UncheckedIcon = styled(FaRegCircle)`
   color: var(--c-gray);
-  transform: scale(2);
+  transform: scale(2.5);
   position: absolute;
```
Comment (ninapeters) [inline frontend/src/Components/DishItem/DishItem.js:66]: The same here.

File: frontend/src/Components/HeaderOverlay/HeaderOverlay.js [hunk 11]
```diff
@@ -0,0 +1,39 @@
+import styled from 'styled-components/macro'
+import PropTypes from 'prop-types'
+import { Link } from 'react-router-dom'
+
+HeaderOverlay.propTypes = {
+  children: PropTypes.string,
```
Comment (ninapeters) [inline frontend/src/Components/HeaderOverlay/HeaderOverlay.js:6]: Isn't it required?

File: frontend/src/Components/HeaderOverlay/HeaderOverlay.js [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+
+const HeaderStyled = styled.header`
+  display: flex;
+  align-items: center;
+  justify-content: space-evenly;
+  z-index: 10;
+  background-color: var(--c-green);
+  box-shadow: 0px 0px 6px 3px rgba(0, 0, 0, 0.08);
+  border-radius: 0 0 15px 15px;
+  position: fixed;
```
Comment (ninapeters) [inline frontend/src/Components/HeaderOverlay/HeaderOverlay.js:26]: Does this work in your styleguide?

File: frontend/src/Components/ListItem/ListItem.js [hunk 13]
```diff
@@ -0,0 +1,40 @@
+import { FaMinusCircle } from 'react-icons/fa'
+import styled from 'styled-components/macro'
+import PropTypes from 'prop-types'
+
+ListItem.propTypes = {
+  id: PropTypes.number,
+  title: PropTypes.string,
+  onDelete: PropTypes.func,
+}
```
Comment (ninapeters) [inline frontend/src/Components/ListItem/ListItem.js:9]: The same with the required.

File: frontend/src/Pages/DishOverviewPage.js [hunk 14]
```diff
@@ -1,46 +1,95 @@
...
+const Grid = styled.div`
+  display: grid;
+  grid-gap: 15px;
+  padding: 20px;
+  grid-row: 2/3;
+`
+
+const LinkOval = styled(Link)`
+  border-radius: 50%;
+  box-shadow: 0px 0px 6px 3px rgba(0, 0, 0, 0.08);
```
Comment (ninapeters) [inline frontend/src/Pages/DishOverviewPage.js:73]: Again the color variable

File: frontend/src/Pages/HomePage.js [hunk 15]
```diff
@@ -0,0 +1,39 @@
+import { Link } from 'react-router-dom'
+import styled from 'styled-components/macro'
+import LogoSmall from '../Assets/Logo_sm.png'
```
Comment (ninapeters) [inline frontend/src/Pages/HomePage.js:3]: I would use lower case for file names.

**Review by ninapeters — COMMENTED** [review]
Very good work. I left a few thoughts and hope they will help you somehow. If something is unclear, feel free to ask.
